ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.cpp

```diff
@@ -239,3 +302,45 @@ void MockVnodes_CheckAndClear()
     
     s_allVnodes.clear();
 }
+
+pair<errno_t, vector<uint8_t>> vnode::ReadXattr(const char* xattrName)
+{
+    // TODO: add support for explicit error mocking
+    XattrMap::const_iterator found = this->xattrs.find(xattrName);
+    if (found == this->xattrs.end())
+    {
+        return make_pair(ENOATTR, vector<uint8_t>());
+    }
+    else
+    {
+        return make_pair(0, found->second);
+    }
+}
+
+SizeOrError Vnode_ReadXattr(vnode_t _Nonnull vnode, const char* _Nonnull xattrName, void* _Nullable buffer, size_t bufferSize)
+{
+    pair<errno_t, vector<uint8_t>> xattrResult = vnode->ReadXattr(xattrName);
+    if (xattrResult.first == 0)
```

Comment:
Making sure I follow here... does .first == 0 mean no error so copy attribute value and return?

Reply:
Yes; perhaps I should define a struct instead of using `pair<>` here.

Reply:
I've solved this with a struct now, please take a look and let me know if that makes it clearer.